test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -49,164 +54,234 @@ class database_operation {
+                collections[collection_id].update_cursor = cursor;
+            }
 
-                /* Start a transaction. */
-                if (!tc->transaction.active())
-                    tc->transaction.begin(tc->session, "");
+            /* Get the random cursor associated with the collection. */
+            cursor = collections[collection_id].random_cursor;
 
-                ts = tc->timestamp_manager->get_next_ts();
-                if (using_timestamps)
-                    tc->transaction.set_commit_timestamp(
-                      tc->session, timestamp_manager::decimal_to_hex(ts));
+            /* Start a transaction. */
+            if (!tc->transaction.active())
+                tc->transaction.begin(tc->session, "");
 
-                update(tc->tracking, cursors[i], collection_names[i], key.c_str(),
-                  generated_value.c_str(), ts);
+            /* Call next to pick a new random record. */
+            ret = cursor->next(cursor);
+            if (ret == WT_NOTFOUND)
+                continue;
+            else if (ret != 0)
+                testutil_die(ret, "unhandled error returned by cursor->next()");
 
-                /* Commit the current transaction. */
-                tc->transaction.op_count++;
-                if (tc->transaction.can_commit())
-                    tc->transaction.commit(tc->session, "");
-            }
-            tc->sleep();
+            /* Get the record's key. */
+            testutil_check(cursor->get_key(cursor, &key_tmp));
+
+            /*
+             * The retrieved key needs to be passed inside the update function. However, the update
+             * API doesn't guarantee our buffer will still be valid once it is called, as such we
+             * copy the buffer and then pass it into the API.
+             */
+            key = key_value_t(key_tmp);
+
+            /* Generate a new value for the record. */
+            generated_value =
+              random_generator::random_generator::instance().generate_string(tc->value_size);
+
+            /*
+             * Get a timestamp to apply to the update. We still do this even if timestamps aren't
+             * enabled as it will return WT_TS_NONE, which is then inserted into the tracking table.
+             */
+            ts = tc->timestamp_manager->get_next_ts();
+            if (using_timestamps)
+                tc->transaction.set_commit_timestamp(
+                  tc->session, timestamp_manager::decimal_to_hex(ts));
+
+            /*
+             * Update the record but take care to handle WT_ROLLBACK as we may conflict with another
+             * running transaction. Here we call the pre-defined wrappers as they also update the
+             * tracking table, which is later used for validation.
+             *
+             * Additionally first get the update_cursor.
+             */
+            cursor = collections[collection_id].update_cursor;
```

Comment:
You can also just keep a handle on the struct instead of looking up every single time like this:
i.e.
```
auto& structThing = collections[collection_id];
foo(structThing.update_cursor);
bar(structThing.random_cursor);
...
```

Reply:
3be345076